ext/sockets/tests/socket_sendto_zerocopy.phpt

```diff
@@ -0,0 +1,47 @@
+--TEST--
+Test socket_sendto with MSG_ZEROCOPY
+--EXTENSIONS--
+sockets
+--SKIPIF--
+<?php
+if (!defined("SO_ZEROCOPY")) {
+    die('skip SO_ZEROCOPY');
+}
+--FILE--
+<?php
+$socket = socket_create(AF_INET, SOCK_DGRAM, SOL_UDP);
+if (!$socket) {
+    die('Unable to create AF_UNIX socket');
+}
+$s = socket_create(AF_INET, SOCK_DGRAM, SOL_UDP);
+if (!$s) {
+    die('Unable to create AF_UNIX socket');
+}
+if (!socket_set_option($socket, SOL_SOCKET, SO_ZEROCOPY, 1)) {
+    die("Unable to set the socket option to SO_ZEROCOPY");
+}
+if (!socket_set_nonblock($s)) {
+    die('Unable to set nonblocking mode for socket');
+}
+$address = '127.0.0.1';
+$port = 3001;
+if (!socket_bind($s, $address, $port)) {
+    die("Unable to bind to $address");
+}
+
+$msg = str_repeat("0123456789abcdef", 1024);
+$len = strlen($msg);
+$bytes_recv = 0;
+$bytes_sent = socket_sendto($socket, $msg, $len, MSG_ZEROCOPY, $address, $port);
+if (socket_recvfrom($s, $resp, 0, MSG_ERRQUEUE, $address, $port) == -1) die ("recvfrom MSG_ERRQUEUE");
+$bytes_recv = socket_recvfrom($s, $resp, 16, 0, $address, $port);
```

Comment:
Isn't this actually sort of async so there's a chance that you need to wait for the buffer to be received first? I'm just wondering if this test could be flaky potentially. I guess it should really wait for the item in error queue for packet originating from SO_EE_ORIGIN_ZEROCOPY, right?